tests/integration/targets/aci_bfd_multihop_interface_profile/tasks/main.yml

```diff
@@ -0,0 +1,174 @@
+    cisco.aci.aci_bfd_multihop_interface_profile:
+      <<: *present_bfd_mh_int_profile
+      state: present
+    register: nm_add_bfd_mh_int_prof
+
+  - name: Verify adding BFD Multihop Interface Profile
+    assert:
+      that:
+      - cm_add_bfd_mh_int_prof is changed
+      - nm_add_bfd_mh_int_prof is changed
+      - cm_add_bfd_mh_int_prof.previous == nm_add_bfd_mh_int_prof.previous == []
+      - cm_add_bfd_mh_int_prof.proposed.bfdMhIfP.attributes.dn == nm_add_bfd_mh_int_prof.proposed.bfdMhIfP.attributes.dn == "uni/tn-ansible_tenant/out-ansible_l3out/lnodep-ansible_node_profile/lifp-ansible_interface_profile/bfdMhIfP"
+      - cm_add_bfd_mh_int_prof.proposed.bfdMhIfP.attributes.keyId == nm_add_bfd_mh_int_prof.proposed.bfdMhIfP.attributes.keyId == "3"
+      - cm_add_bfd_mh_int_prof.proposed.bfdMhIfP.attributes.type == nm_add_bfd_mh_int_prof.proposed.bfdMhIfP.attributes.type == "none"
+
+  - name: Add BFD Multihop Interface Profile again- idempotency
+    cisco.aci.aci_bfd_multihop_interface_profile:
+      <<: *present_bfd_mh_int_profile
+      state: present
+    register: add_bfd_mh_int_prof_again
+
+  - name: Verify adding BFD Multihop Interface Profile again - idempotency
+    assert:
+      that:
+      - add_bfd_mh_int_prof_again is not changed
+
+  # CREATE BFD Multihop Interface policy
+  - name: Add a new  BFD Multihop Interface policy
+    cisco.aci.aci_bfd_multihop_interface_policy:
+      <<: *aci_info
+      tenant: ansible_tenant
+      name: ansible_bfd_multihop_interface_policy
+      description: Ansible BFD Multihop Interface Policy
+      state: present
+    register: add_bfd_multihop_interface_pol
+
+  # Update BFD Multihop Interface Profile
+  - name: Update BFD Multihop Interface Profile
+    cisco.aci.aci_bfd_multihop_interface_profile:
+      <<: *present_bfd_mh_int_profile
+      interface_profile_type: sha1
+      key: "abc*123"
+      key_id: 15
+      bfd_multihop_interface_policy: ansible_bfd_multihop_interface_policy
+      state: present
+    register: update_bfd_mh_int_prof
+
+  - name: Verify updating BFD Multihop Interface Profile
+    assert:
+      that:
+      - update_bfd_mh_int_prof is changed
+      - update_bfd_mh_int_prof.sent.bfdMhIfP.attributes.keyId == "15"
+      - update_bfd_mh_int_prof.sent.bfdMhIfP.attributes.key == "abc*123"
+      - update_bfd_mh_int_prof.sent.bfdMhIfP.attributes.type == "sha1"
+      - update_bfd_mh_int_prof.sent.bfdMhIfP.children.0.bfdRsMhIfPol.attributes.tnBfdMhIfPolName == "ansible_bfd_multihop_interface_policy"
+
+  - name: Query BFD Multihop Interface Profile
+    cisco.aci.aci_bfd_multihop_interface_profile:
+      <<: *present_bfd_mh_int_profile
+      state: query
+    register: query_bfd_mh_int_prof
+
+  - name: Query all BFD Multihop Interface Profile
+    cisco.aci.aci_bfd_multihop_interface_profile:
+      <<: *aci_info
+      state: query
+    register: query_all_bfd_mh_int_prof
+
+  - name: Delete BFD Multihop Interface Profile - check mode
```

Comment:
is there reason to only execute checkmode for delete?

Reply:
added normal mode and idempotency